src/runtime/gravity_core.c

```diff
@@ -2058,6 +2058,70 @@ static bool string_split (gravity_vm *vm, gravity_value_t *args, uint16_t nargs,
 	RETURN_VALUE(VALUE_FROM_OBJECT(list), rindex);
 }
 
+static bool string_loop (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	if (nargs < 2) RETURN_ERROR("Incorrect number of arguments.");
+	if (!VALUE_ISA_CLOSURE(GET_VALUE(1))) RETURN_ERROR("Argument must be a Closure.");
+
+	gravity_closure_t *closure = VALUE_AS_CLOSURE(GET_VALUE(1));	// closure to execute
+	gravity_value_t value = GET_VALUE(0);							// self parameter
+	gravity_string_t *string = VALUE_AS_STRING(value);
+	char *str = string->s;
+	register gravity_int_t n = string->len;  // Times to execute the loop
+	register gravity_int_t i = 0;
+
+	gravity_list_t *list = gravity_list_new(vm, 0);
+	for (int i = 0; i <= n; ++i) {
+		// A litte trick to make a 1 character string so that VALUE_FROM_STRING will
+		// accept it.
+		char c[1] = {str[i]};
+		marray_push(gravity_value_t, list->array, VALUE_FROM_STRING(vm, c, 1));
```

Comment:
Instead of creating an array of 1 you could just create one char and get the pointer.
```
char c = str[i];
marray_push(gravity_value_t, list->array, VALUE_FROM_STRING(vm, &c, 1));
```

Or even better with an little bit of pointer arithmetic you could accomplish the same thing without creating an temporary variable. Although I'm not very familiarized with C so I don't know if it's valid or a good thing to do.

```
marray_push(gravity_value_t, list->array, VALUE_FROM_STRING(vm, str + i, 1));
```

Reply:
Nice, didn't think of that one. Your second one does work.